### ISSUE TITLE
Custom logging implementation

### ISSUE THREAD

**Issue by timkoers** [square/picasso#1190]
Hi everyone,

Recently I started logging a lot more in my application to get a clearer view of what could possibly be crashing the app. I used Crashlytics for this, this works excellent, but there is one drawback of combining Picasso and Crashlytics: You can't log Picasso to Crashlytics.

Can you make some kind of interface that sends a callback everytime a new log is logged?

That would be very helpful to understand what Picasso is actually doing when it doesn't show images.

**Comment by JakeWharton** [comment]
Picasso logging is both extremely verbose and extremely expensive in terms of overhead. We do not provide a logging interface because having debug logs enabled in your production app is a bad idea and will cause noticeable performance degradation.

Unless there's another reason to do this, I can't see any reason to add this feature.

**Comment by arjit-meesho** [comment]
@JakeWharton We recently faced a problem where images were not loading on a user's device and we were not able to reproduce the issue. We have a custom file logging implementation for Android. Since, `okHttp` provide us with custom logger we were able to receive `okhttp` logs in our file logs. We wanted to ship him another APK so that we can read picasso logs too.

Although we have disabled picasso logs on production app but this could be helpful for easier debugging of remote failures.

I think this is a valid use case. It would be better if Picasso can either provide us with file logs or an option to add custom logger.
